uk.co.compucorp.civicrm.hrcore/backstop_data/page-objects/ssp-la-my-leave.js

```diff
@@ -0,0 +1,44 @@
+var Promise = require('es6-promise').Promise;
+var page = require('./page');
+
+var currentMonth = (new Date()).getMonth() + 1;
+
+module.exports = (function () {
+
+  return page.extend({
+    /**
+     * Wait for current month to be visible
+     */
+    waitForCurrentMonthsVisibility: function () {
+      var casper = this.casper;
+
+      casper.then(function () {
+        //Wait for the current month spinner to be first visible then hidden
+        casper.waitUntilVisible('.panel.panel-default:nth-of-type(' + currentMonth + ') ng-include .chr_spinner', function () {
+          casper.waitWhileVisible('.panel.panel-default:nth-of-type(' + currentMonth + ') ng-include .chr_spinner');
+        });
+      });
+    },
+    /**
+     * Clear the months selected to show all months
+     */
+    showAllMonths: function () {
+      var casper = this.casper;
+      //If Current month is december then need to wait for November as it would be loaded last when all months load
+      var lastMonth = currentMonth === 12 ? 11 : 12;
+
+      casper.then(function () {
+        //Wait for the current month spinner to be first visible then hidden
+        casper.waitUntilVisible('.panel.panel-default:nth-of-type(' + currentMonth + ') ng-include .chr_spinner', function () {
+          casper.waitWhileVisible('.panel.panel-default:nth-of-type(' + currentMonth + ') ng-include .chr_spinner', function () {
+            casper.click('.panel-heading .close.ui-select-match-close');
+            //Wait for the last month spinner to be first visible then hidden
+            casper.waitUntilVisible('.panel.panel-default:nth-of-type(' + lastMonth + ') ng-include', function () {
```

Comment:
if this "wait for it to be visible, then to be hidden" logic cannot be avoided, then please put it in a private function, cause you have lots of repetition with it